### PR TITLE
Add experimental Docker Stack CLI commands

### PR DESCRIPTION
**Pull request by dnephin**
Adds experimental stack commands for deploying a bundle

### CODE REVIEW

**Comment by duglin** [comment]
@dnephin where can we read up on this feature?

**Comment by icecrime** [comment]
@duglin I'm going to try and write a doc page today, sorry for the delay!

**Comment by mlaventure** [comment]
Can we have `// +build experimental` added to all the new stack files to avoid compiling them when building the non experimental version?

**Comment by dnephin** [comment]
@mlaventure done

File: api/client/stack/deploy.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		return err
+	}
+	return deployServices(ctx, dockerCli, bundle.Services, opts.namespace)
+}
+
+func getUniqueNetworkNames(services map[string]bundlefile.Service) []string {
+	networkSet := make(map[string]bool)
+	for _, service := range services {
+		for _, network := range service.Networks {
+			networkSet[network] = true
```
Comment (mlaventure) [inline api/client/stack/deploy.go]: can't we just append those in the `networks []string` variable from the get go here?

I'm not sure what is the benefit of going through an intermediate map first.

Comment (mlaventure) [inline api/client/stack/deploy.go]: Scratch that, I read the function name afterwards :)

Comment (dnephin) [inline api/client/stack/deploy.go]: Ya, I don't know if there is a better way to do this in go.

**Comment by mlaventure** [comment]
LGTM (IANAM)

Only nitpick I've got is:
```
func getNetworks(
 	ctx context.Context,
 	apiclient client.APIClient,
 	namespace string,
 ) ([]types.NetworkResource, error) {
 	return apiclient.NetworkList(
 		ctx,
 		types.NetworkListOptions{Filters: getStackFilter(namespace)})
 }
```

we have no other code with the above style as we don't enforce the 80 columns rule

**Comment by vieux** [comment]
LGTM

**Comment by icecrime** [comment]
Added some docs in #23544.

**Comment by icecrime** [comment]
LGTM :+1:

**Comment by vieux** [comment]
Mush better with some docs thanks @icecrime 

LGTM

**Comment by cpuguy83** [comment]
LGTM moving to docs

**Comment by icecrime** [comment]
So @thaJeztah is on a plane, and this is experimental, soooooooo... :innocent:

**Pull request merged**